packages/shared/src/components/squads/moderation/SquadModerationList.tsx

```diff
@@ -20,9 +23,17 @@ export function SquadModerationList({
   const moderate = useSourceModerationList({
     squad,
   });
-  const { data, isFetched, fetchNextPage, hasNextPage, isPending } =
-    useSquadPendingPosts(squad?.id);
   const isModerator = verifyPermission(squad, SourcePermissions.ModeratePost);
+  const { data, isFetched, fetchNextPage, hasNextPage, isPending } =
+    useSquadPendingPosts(
+      squad?.id,
+      isModerator
+        ? [SourcePostModerationStatus.Pending]
+        : [
+            SourcePostModerationStatus.Pending,
+            SourcePostModerationStatus.Rejected,
+          ],
+    );
```

Comment:
`status` is not included in the query key so this will lead to a bug where pending/rejected posts will show the same data depending on which one is queried first.

Reply:
cc @AmarTrebinjac